app/src/display/widgets/peripheral_status.c

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) 2022 The ZMK Contributors
+ *
+ * SPDX-License-Identifier: MIT
+ */
+
+#include <kernel.h>
+#include <bluetooth/services/bas.h>
+
+#include <logging/log.h>
+LOG_MODULE_DECLARE(zmk, CONFIG_ZMK_LOG_LEVEL);
+
+#include <zmk/display.h>
+#include <zmk/display/widgets/peripheral_status.h>
+#include <zmk/event_manager.h>
+#include <zmk/split/bluetooth/peripheral.h>
+#include <zmk/events/split_peripheral_status_changed.h>
+
+static sys_slist_t widgets = SYS_SLIST_STATIC_INIT(&widgets);
+
+struct peripheral_status_state {
+    bool connected;
+};
+
+static struct peripheral_status_state get_state(const zmk_event_t *_eh) {
+    return (struct peripheral_status_state){.connected = zmk_split_bt_peripheral_is_connected()};
+}
+
+static void set_status_symbol(lv_obj_t *label, struct peripheral_status_state state) {
+    char *text =
```

Comment:
This should probably be a const pointer. Otherwise you'd be allowed to try to modify a constant string.